packages/engine/docker/README.md

```diff
@@ -0,0 +1,36 @@
+# Docker containers for hEngine
+
+This directory contains the Docker images for hEngine. To build the image, run from the `packages/engine` directory:
+
+```sh
+docker build -t h-engine -f docker/Dockerfile .
+```
+
+To run it, a project has to be passed as volume mount:
+
+```sh
+docker run -v "$(pwd)/tests/examples/air_defense_system:/air_defense_system:ro" h-engine \
+  --project /air_defense_system \
+  --log-level info \
+  single-run \
+  --num-steps 10
+```
+
+It's possible to set the output as well:
+
+```sh
+mkdir -p output
+
+docker run -v "$(pwd)/tests/examples/air_defense_system:/air_defense_system:ro" -v "$(pwd)/output:/output" h-engine \
+  --project /air_defense_system \
+  --log-level info \
+  --output /output \
+  single-run \
+  --num-steps 10
+```
+
+Eventually, a permission denied error will occur. By default, the Docker image uses the user `1000:1000`. If you have another user id, you probably need to rebuild the image with your id. To add a user with the correct id and run the container again:
```

Comment:
I'm a little unclear as to what this is saying. Will you always have to rebuild the image with your id? Why does this happen "eventually"?

Reply:
Docker behaves weirdly with users and permissions. You only need to rebuild the image once.
```suggestion
Eventually, a permission denied error will occur when attempting to write to the output folder. This is because the Docker image uses `1000:1000` as user. It's possible to change the user inside of the docker image by rebuilding it with another id:
```

Reply:
Gotcha, the phrasing of this sounds like the intended approach is to use the docker image, set the output directory, and then run it where it _will_ fail.. and only then fix it.. why not just get them to set the user before setting the output directory?

Reply:
Because it does not necessarily fail. `1000:1000` is the default user for almost all unix distributions. Unless you have a multi-user setup, this is likely to pass.

Reply:
"Eventually" implies it will happen at some point, rather than it _may_ 😁 

```suggestion
The Docker image uses `1000:1000` as the user, because of this you may encounter a permission denied error when attempting to write to the output folder. It's possible to fix this by changing the user inside of the docker image by rebuilding it with another id:
```

Do we perhaps also want to give the command to get your user id?

Reply:
```suggestion
The Docker image uses 1000:1000 as the user by default. Because of this you may encounter a permission denied error when attempting to write to the output folder. It's possible to fix this by changing the user inside of the docker image by rebuilding it with another id:
```

The command for getting the user id and group id is inside the command (`id -u` and `id -g`). Do you think we still should explicitly state that?

Reply:
Oh my mistake I missed it. It's probably fine